src/librustc_mir/borrow_check/mod.rs

```diff
@@ -464,13 +464,7 @@ impl<'cx, 'gcx, 'tcx> DataflowResultsConsumer<'cx, 'tcx> for MirBorrowckCtxt<'cx
                 target: _,
                 unwind: _,
             } => {
-                self.access_place(
-                    ContextKind::Drop.new(loc),
-                    (drop_place, span),
-                    (Deep, Write(WriteKind::StorageDeadOrDrop)),
-                    LocalMutationIsAllowed::Yes,
-                    flow_state,
-                );
+                self.visit_terminator_drop(loc, term, flow_state, drop_place, span);
```

Comment:
Here, we will start by erasing the type of the `drop_place`, roughly like so:

```rust
let gcx = self.tcx.global_tcx();

// Compute the type with accurate region information.
let drop_place_ty = drop_place.ty(self.mir, self.tcx);

// Erase the regions.
let drop_place_ty = self.tcx.erase_regions(&drop_place_ty);

// "Lift" into the gcx -- once the regions are erased, this type should be in the global arenas; this "lift" operation basically just asserts that this is true, but that's useful later.
let drop_place_ty = gcx.lift(&drop_place_ty).unwrap();
```

(Note that I am intentionally shadowing `drop_place_ty` here, because we don't want to accidentally access the intermediate values.)

Then we'll add this `drop_place_ty` as an argument to `visit_terminator_drop`